### PR TITLE
(FACT-2030) Cleanup resolvers

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: lib/facts/debian/os/distro.rb [hunk 1]
```diff
@@ -7,9 +7,9 @@ class OsLsbRelease
...
       def call_the_resolver
         distro = {
-          'codename' => resolver('Codename'),
-          'description' => resolver('Description'),
-          'id' => resolver('Distributor ID'),
+          'codename' => resolver(:codename),
+          'description' => resolver(:description),
+          'id' => resolver(:distributor_id),
           'release' => {
             'full' => Resolvers::DebianVersion.resolve(:full),
```
Comment (IrimieBogdan) [inline lib/facts/debian/os/distro.rb]: Would be nice to extract the resolver in a different method, like in line 12.

**Review by BogdanIrimie — APPROVED** [review]

File: spec/facter/resolvers/utils/windows/win32ole_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    allow(result).to receive(:ConnectServer).with('.', 'root\\cimv2').and_return(result)
+    allow(result).to receive(:Security_).and_return(result)
+    allow(result).to receive(:ImpersonationLevel=).and_return(result)
+    allow(result).to receive(:execquery).with(query).and_return(query_result)
+  end
+
+  context '#return_first returns nil' do
+    let(:query) { 'query' }
+    let(:query_result) {}
+    it '' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/utils/windows/win32ole_spec.rb]: Don't leave empty **it** statements

File: spec/facter/resolvers/aix/architecture_resolver_spec.rb [hunk 3]
```diff
@@ -20,6 +20,13 @@
     end
   end
 
+  context '#resolve' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/aix/architecture_resolver_spec.rb]: There is already a context `#resolve` defined

File: lib/resolvers/os_release_resolver.rb [hunk 4]
```diff
@@ -33,9 +33,8 @@ def read_os_release_file(fact_name)
             pairs << line.strip.delete('"').split('=', 2)
           end
 
-          @fact_list = Hash[*pairs.flatten]
-
-          @fact_list[:identifier] = @fact_list['ID'].downcase
```
Comment (sebastian-miclea) [inline lib/resolvers/os_release_resolver.rb]: `:identifier` is used by the `CurrentOs` class

File: lib/resolvers/lsb_release_resolver.rb [hunk 5]
```diff
@@ -24,8 +24,8 @@ def read_lsb_release_file(fact_name)
           output, _status = Open3.capture2('lsb_release -a')
           release_info = output.delete("\t").split("\n").map { |e| e.split(':') }
 
-          @fact_list = Hash[*release_info.flatten]
-          @fact_list[:identifier] = @fact_list['Distributor ID'].downcase
```
Comment (sebastian-miclea) [inline lib/resolvers/lsb_release_resolver.rb]: `:identifier` is used by the `CurrentOs` class

File: lib/resolvers/sw_vers_resolver.rb [hunk 6]
```diff
@@ -22,7 +22,8 @@ def resolve(fact_name)
         def software_version_system_call(fact_name)
           output, _status = Open3.capture2('sw_vers')
           release_info = output.delete("\t").split("\n").map { |e| e.split(':') }
-          @fact_list = Hash[*release_info.flatten]
+          result = Hash[*release_info.flatten]
+          result.each { |k, v| @fact_list[k.downcase.to_sym] = v }
           @fact_list[fact_name]
```
Comment (sebastian-miclea) [inline lib/resolvers/sw_vers_resolver.rb]: Isn't `@fact_list` empty?

File: spec/facter/resolvers/windows/win_os_release_resolver_spec.rb [hunk 7]
```diff
@@ -136,4 +136,15 @@
...
     end
   end
+
+  context '#resolve version is 4.2' do
+    let(:comp) { double('Win32Ole', ProductType: prod, OtherTypeDescription: type) }
+    let(:version) { '4.2' }
+    let(:prod) { '2' }
+    let(:type) { 'R2' }
+
+    it 'facts are nil' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/windows/win_os_release_resolver_spec.rb]: The test says that they are `4.2`, not `nil`.

File: spec/facter/resolvers/hostname_resolver_spec.rb [hunk 8]
```diff
@@ -5,11 +5,23 @@
...
   context '#resolve' do
     let(:host) { 'hostname' }
 
-    it 'detects that hostname is nil' do
+    it 'detects hostname' do
       expect(Facter::Resolvers::Hostname.resolve(:hostname)).to eql('hostname')
     end
   end
+
+  context '#resolve' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/hostname_resolver_spec.rb]: Same context defined twice.

File: lib/resolvers/sw_vers_resolver.rb (hunk 6 above)
Comment (oanatmaria) [reply lib/resolvers/sw_vers_resolver.rb]: @fact_list[k.downcase.to_sym] = v , in each @fact_list is populated

File: spec/facter/resolvers/aix/architecture_resolver_spec.rb [hunk 9]
```diff
@@ -20,9 +20,16 @@
     end
   end
 
+  context '#resolve when query result does not contain "value"' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/aix/architecture_resolver_spec.rb]: I would like to propose a structure for tests to have them easier to understand and find:
```
describe 'ClassName' do
  context '#method' do
    # here you can have common stubs for that #method requires
    context 'test path1' do
      # here you can have specific test path stubs
      it 'responds x' do end
    end

    context 'test path2' do
      it 'responds y' do end
    end
  end

  context '#method2' do
    context 'test path1' do # optional
      it ...
      it ...
    end
  end
end
```
@oanatmaria @IrimieBogdan @loopinu what do you think?

Comment (IrimieBogdan) [reply spec/facter/resolvers/aix/architecture_resolver_spec.rb]: If the test are complex, I am for this structure. I would apply it for new tests and not rewrite existing ones.

**Review by sebastian-miclea — APPROVED** [review]

**Pull request merged**